.travis/build.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e # exit with nonzero exit code if anything fails
+
+env
+unset SPIFFY
+unset ESPTOOL2
+
+export SMING_HOME=$TRAVIS_BUILD_DIR/Sming
+export ESP_HOME=$TRAVIS_BUILD_DIR/opt/esp-alt-sdk
+if [ "$SDK_VERSION" = "3.0.0" ]; then
+  export SDK_BASE=$SMING_HOME/third-party/ESP8266_NONOS_SDK
+fi
+
+cd $SMING_HOME
+
+export PATH=$PATH:$ESP_HOME/xtensa-lx106-elf/bin:$ESP_HOME/utils/:$SMING_HOME/../.travis/tools
+if [ "$SDK_VERSION" = "2.0.0" ]; then
+  ../.travis/tools/clang/format-pr.sh;
+fi
+
+if [ "$SDK_VERSION" = "3.0.0" ]; then
```

Comment:
Make the `Host` CI checks a separated build in the matrix. Don't smuggle it inside the 3.0.0 SDK build.

Reply:
Agreed. I also think that running everything in parallel is inefficient and wasteful of resources. I'd like to propose we break it into stages, running all our unit tests first before proceeding to platform builds. I've managed to get Travis doing a staged build for us, like this:

https://travis-ci.org/mikee47/Sming/builds/538921558

As it's using native compilers the test stage completes very quickly.

Any thoughts?

Reply:
Also, is it worth (attempting to) moving Windows builds onto Travis? Any particular reason we need both?

Reply:
> I'd like to propose we break it into stages, running all our unit tests first before proceeding to platform builds.

Sounds good to me.

> Also, is it worth (attempting to) move Windows builds onto Travis? Any particular reason we need both it and Appveyor?

Travis was providing only Linux and Mac OS boxes. If nowadays it provides also Windows boxes then I have nothing against using Travis instead of Appveyor for the Windows builds.

Reply:
> I like the staged build approach. Let's go for it.

The coding style rule checker can be moved to a separate stage/job. We can have something line
1. Check coding style.
2. Run (unit)tests.

3. Run Arch builds in parallel.

Reply:
I'll put the coding style checks in the initial test stage, no point creating a separate VM just for that